Makes/MakesViewController.m

```diff
@@ -0,0 +1,111 @@
+//
+//  ViewController.m
+//  Makes
+//
+//  Created by Mohamed Ragab on 1/8/17.
+//  Copyright © 2017 Mohamed Ragab. All rights reserved.
+//
+
+#import "MakesViewController.h"
+#import "Make.h"
+#import <AFNetworking/AFNetworking.h>
+#import <Masonry/Masonry.h>
+#import "MakesCollectionViewCell.h"
+#import "MakeDetailViewController.h"
+#import "MakesManager.h"
+#import <SDWebImage/UIImageView+WebCache.h>
+
+
+
+
+@interface MakesViewController ()
+@property (nonatomic, strong) NSArray *makes;
+@property (nonatomic, strong) UICollectionView *makesCollectionView;
+@end
+
+@implementation MakesViewController
+
+- (void)loadMakes {
+    
+    [[MakesManager manager] makesWithsuccess:^(NSArray *makes) {
+        self.makes = makes;
+        [self.makesCollectionView reloadData];
+    } failure:^(NSError *error) {
+        
+    }];
+
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    self.makes = [[NSMutableArray alloc]init];
```

Comment:
look above we set makes array to be NSArray how do you allocate it with NSMutableArray also why you init it ? we already used it and it's initialized with the makes array in loadMakes method